homeassistant/components/sensibo/coordinator.py

```diff
@@ -0,0 +1,120 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, FIELD_TO_FLAG, LOGGER, TIMEOUT
+
+
+class SensiboDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Sensibo Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Sensibo coordinator."""
+        self.client = pysensibo.SensiboClient(
+            entry.data[CONF_API_KEY],
+            session=async_get_clientsession(hass),
+            timeout=TIMEOUT,
+        )
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[str, dict[str, Any]]:
+        """Fetch data from Sensibo."""
+
+        devices = []
+        try:
+            for dev in await self.client.async_get_devices():
+                devices.append(dev)
+        except (
+            ClientConnectionError,
+            asyncio.TimeoutError,
+            pysensibo.SensiboError,
+        ) as error:
+            raise UpdateFailed from error
+
+        entities: dict[str, dict[str, Any]] = {}
+        for dev in devices:
+            unique_id = dev["id"]
+            name = dev["room"]["name"]
+            temperature = dev["measurements"].get("temperature", 0)
+            humidity = dev["measurements"].get("humidity", 0)
+            ac_states = dev["acState"]
+            target_temperature = ac_states.get("targetTemperature")
+            hvac_mode = ac_states.get("mode")
+            running = ac_states.get("on")
+            fan_mode = ac_states.get("fanLevel")
+            swing_mode = ac_states.get("swing")
+            available = dev["connectionStatus"].get("isAlive", True)
+            capabilities = dev["remoteCapabilities"]
+            hvac_modes = list(capabilities["modes"])
+            if hvac_modes:
+                hvac_modes.append("off")
+            current_capabilities = capabilities["modes"][ac_states.get("mode")]
+            fan_modes = current_capabilities.get("fanLevels")
+            swing_modes = current_capabilities.get("swing")
+            temperature_unit_key = dev.get("temperatureUnit") or ac_states.get(
+                "temperatureUnit"
+            )
+            temperatures_list = (
+                current_capabilities["temperatures"]
+                .get(temperature_unit_key, {})
+                .get("values", [0])
+            )
+            if temperatures_list:
+                temperature_step = temperatures_list[1] - temperatures_list[0]
+            supported_features = 0
+            for key in ac_states:
+                if key in FIELD_TO_FLAG:
+                    supported_features |= FIELD_TO_FLAG[key]
```

Comment:
The coordinator shouldn't know about platform details like supported features. Keep that in the platform.